packages/office-ui-fabric-react/src/components/Slider/Slider.tsx

```diff
@@ -175,31 +183,56 @@ export class Slider extends BaseComponent<ISliderProps, ISliderState> implements
     const { max, min, step } = this.props;
     const steps: number = (max! - min!) / step!;
     const sliderPositionRect: ClientRect = this.refs.sliderLine.getBoundingClientRect();
-    const sliderLength: number = sliderPositionRect.width;
+    const sliderLength: number = !this.props.vertical ? sliderPositionRect.width : sliderPositionRect.height;
     const stepLength: number = sliderLength / steps;
     let currentSteps: number | undefined;
-
-    if (event.type === 'mousedown' || event.type === 'mousemove') {
-      currentSteps = getRTL() ?
-        (sliderPositionRect.right - (event as MouseEvent).clientX) / stepLength :
-        ((event as MouseEvent).clientX - sliderPositionRect.left) / stepLength;
-    } else if (event.type === 'touchstart' || event.type === 'touchmove') {
-      currentSteps = getRTL() ?
-        (sliderPositionRect.right - (event as TouchEvent).touches[0].clientX) / stepLength :
-        ((event as TouchEvent).touches[0].clientX - sliderPositionRect.left) / stepLength;
+    if (!this.props.vertical) {
+      if (event.type === 'mousedown' || event.type === 'mousemove') {
+        currentSteps = getRTL() ?
+          (sliderPositionRect.right - (event as MouseEvent).clientX) / stepLength :
+          ((event as MouseEvent).clientX - sliderPositionRect.left) / stepLength;
+      } else if (event.type === 'touchstart' || event.type === 'touchmove') {
+        currentSteps = getRTL() ?
+          (sliderPositionRect.right - (event as TouchEvent).touches[0].clientX) / stepLength :
+          ((event as TouchEvent).touches[0].clientX - sliderPositionRect.left) / stepLength;
+      }
+    } else {
+      if (event.type === 'mousedown' || event.type === 'mousemove') {
+        currentSteps = getRTL() ?
```

Comment:
duplicate code... can you please simplify this? The only difference is is a few X/Y values. Maybe make a function which takes these as input params and call that, to avoid duplication.

Reply:
Hi, @dzearing , I'm not sure if this will meet the requirements. Can you review it again? Thanks.